packages/algob/src/lib/algo-sdk.ts

```diff
@@ -0,0 +1,67 @@
+import algosdk from "algosdk";
+
+import {
+  Account,
+  ASADef,
+  ASADeploymentFlags,
+  ASAInfo
+} from "../types";
+import * as t from "./tx";
+
+const confirmedRound = "confirmed-round";
+
+export interface AlgoSDKWrapper {
+  deployASA: (name: string, asaDesc: ASADef, flags: ASADeploymentFlags, account: Account) => Promise<ASAInfo>
+}
+
+export class AlgoSDKWrapperDryRunImpl implements AlgoSDKWrapper {
+  async deployASA (
+    name: string, asaDesc: ASADef, flags: ASADeploymentFlags, account: Account
+  ): Promise<ASAInfo> {
+    return {
+      creator: account.addr + "-get-address-dry-run",
+      txId: "tx-id-dry-run",
+      assetIndex: -1,
+      confirmedRound: -1
+    };
+  }
+}
+
+export class AlgoSDKWrapperImpl implements AlgoSDKWrapper {
+  private readonly algoClient: algosdk.Algodv2;
+
+  constructor (algoClient: algosdk.Algodv2) {
+    this.algoClient = algoClient;
+  }
+
+  // Source:
+  // https://github.com/algorand/docs/blob/master/examples/assets/v2/javascript/AssetExample.js#L21
+  // Function used to wait for a tx confirmation
+  async waitForConfirmation (txId: string): Promise<algosdk.PendingTransactionInformation> {
+    const response = await this.algoClient.status().do();
+    let lastround = response["last-round"];
+    while (true) {
+      const pendingInfo = await this.algoClient.pendingTransactionInformation(txId).do();
+      if (pendingInfo[confirmedRound] !== null && pendingInfo[confirmedRound] > 0) {
+        return pendingInfo;
+      }
+      lastround++;
+      await this.algoClient.statusAfterBlock(lastround).do();
+    }
+  };
+
+  async deployASA (
+    name: string, asaDesc: ASADef, flags: ASADeploymentFlags, account: Account
+  ): Promise<ASAInfo> {
+    const tx = await t.makeAssetCreateTxn(this.algoClient, asaDesc, flags, account);
+    const rawSignedTxn = tx.signTxn(account.sk);
+    const txInfo = await this.algoClient.sendRawTransaction(rawSignedTxn).do();
+    const txConfirmation = await this.waitForConfirmation(txInfo.txId);
+    return {
+      creator: account.addr,
+      txId: txInfo.txId,
+      assetIndex: txConfirmation["asset-index"],
+      confirmedRound: txConfirmation[confirmedRound]
+    };
+  }
+}
```

Comment:
So that I could mock it without mocking all of `Algodv2's` methods.

Reply:
We don't do it in JS / TypeScript. There are other (better) ways to do it:

1. You can override a method, eg:
        let algod = new algosdk.Algodv2
        algod.methodToOverrid = (...) => {... }
2. You can use a mocking library. We are already using `sinon` for that. You can also check this [article](https://www.stackabuse.com/using-mocks-for-testing-in-javascript-with-sinon-js/) for examples.

Reply:
I still want to keep the class (name is not important, we can change it) so that Deployer would offload all of the node interaction logic to it and won't have any private methods.